src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -115,20 +133,43 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
      ********************************************************************************************************************/
     public override DmlWrapper run(List<SObject> newSObjects, List<SObject> oldSObjects,
         TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+        if (triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
+            hasRun = false;
+
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.HH, false);
+        }
+        if (triggerAction == TDTM_Runnable.Action.BeforeInsert
+                || triggerAction ==TDTM_Runnable.Action.AfterInsert) {
+            isNewHouseholdAccountInsert = true;
+            isNewContactInsert = true;
+            hasRun = true;
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.HH, true);
+        }
+
+        // Reset this flag to false for each call to run(), since some NPSP tests
+        // insert Contacts in multiple DML statements.  This flag functions then as
+        // protection against recursion for each call to the method.
+        isNewHouseholdAccountInsert = false;
 
         this.contacts = newSObjects;
         this.oldContactById = oldSObjects == null ? null : new Map<Id, Contact>((List<Contact>) oldSObjects);
         this.triggerAction = triggerAction;
         this.dmlWrapper = new DmlWrapper();
         contactsSettings = UTIL_CustomSettingsFacade.getContactsSettings();
 
+        if (triggerAction == TDTM_Runnable.Action.BeforeInsert
+                || triggerAction == TDTM_Runnable.Action.AfterInsert) {
+            isNewContactInsert = true;
```

Comment:
@kenlewis-sfdo just noticed - is this assignment redundant? Isn't it being assigned w/in an equivalent condition about a dozen lines up?